docs/src/index.md

```diff
@@ -5,11 +5,109 @@
 
 
 [ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl) helps you test [`ChainRulesCore.frule`](http://www.juliadiff.org/ChainRulesCore.jl/dev/api.html) and [`ChainRulesCore.rrule`](http://www.juliadiff.org/ChainRulesCore.jl/dev/api.html) methods, when adding rules for your functions in your own packages.
-
 For information about ChainRules, including how to write rules, refer to the general ChainRules Documentation:
 [![](https://img.shields.io/badge/docs-master-blue.svg)](https://JuliaDiff.github.io/ChainRulesCore.jl/dev)
 [![](https://img.shields.io/badge/docs-stable-blue.svg)](https://JuliaDiff.github.io/ChainRulesCore.jl/stable)
 
+## Canonical example
+
+Let's suppose a custom transformation has been defined
+```
+function two2three(a::Float64, b::Float64)
+    return 1.0, 2.0*a, 3.0*b
+end
+```
+along with the `frule`
+```
+function ChainRulesCore.frule((Δf, Δa, Δb), ::typeof(two2three), a, b)
+    y = two2three(a, b)
+    ∂y = Composite{Tuple{Float64, Float64, Float64}}(Zero(), 2.0*Δa, 3.0*Δb)
+    return y, ∂y
+end
+```
+and `rrule`
+```
+function ChainRulesCore.rrule(::typeof(two2three), a, b)
+    y = two2three(a, b)
+    function two2three_pullback(Ȳ)
+        return (NO_FIELDS, 2.0*Ȳ[2], 3.0*Ȳ[3])
+    end
+    return y, two2three_pullback
+end
+```
+
+The `test_frule`/`test_rrule` helper function compares the `frule`/`rrule` outputs
+to the gradients obtained by finite differencing.
+They can be used for any type and number of inputs and outputs.
+
+### Testing the `frule`
+
+`frule_test` takes in the function `f` and tuples `(x, ẋ)` for each function argument `x`.
+The call will test the `frule` for function `f` at the point `x` in the domain. Keep
+this in mind when testing discontinuous rules for functions like
+[ReLU](https://en.wikipedia.org/wiki/Rectifier_(neural_networks)), which should ideally
+be tested at both `x` being above and below zero.
+Additionally, choosing `ẋ` in an unfortunate way (e.g. as zeros) could hide
+underlying problems with the defined `frule`.
+
+```
+xs = (3.33, -7.77)
+ẋs = (rand(), rand())
+frule_test(two2three, (xs[1], ẋs[1]), (xs[2], ẋs[2]))
+```
+
+### Testing the `rrule`
+
+`rrule_test` takes in the function `f`, sensitivities of the function outputs `ȳ`,
+and tuples `(x, x̄)` for each function argument `x`.
+`x̄` is the accumulated adjoint which should be set randomly.
+The call will test the `rrule` for function `f` at the point `x`, and similarly to
+`frule` some rules should be tested at multiple points in the domain.
+Choosing `ȳ` in an unfortunate way (e.g. as zeros) could hide underlying problems with
+the `rrule`. 
+```
+xs = (3.33, -7.77)
+ȳs = (rand(), rand(), rand())
+x̄s = (rand(), rand())
```

Comment:
similar for my comments about maybe we avoid using `xs` and just use `a` and `b` seperately

Reply:
I went with `x1` and `x2` (changing the function definition to be consistent instead) just because the text feels more natural talking about arguments `x` rather than `a`